tests/cypress/e2e/unit_tests/rbac_fleet.spec.ts

```diff
@@ -579,13 +586,63 @@ describe('Test GitRepoRestrictions scenarios for GitRepo applicaiton deployment.
       cy.filterInSearchBox(appName);
       cy.get('.col-link-detail').contains(appName).should('be.visible');
 
+      // Deleting GitRepoRestrictions from the fleet-local namespace
+      cy.accesMenuSelection('Continuous Delivery', 'Advanced', 'GitRepoRestrictions');
+      cy.fleetNamespaceToggle('fleet-local');
+      cy.deleteAll(false);
+
       // Delete GitRepo
       cy.deleteAllFleetRepos();
+    })
+  )
+
+  qase(41,
+    it('Test "allowedTargetNamespace" from "GitRepoRestrictions" overrides "defaultNamespace" in fleet.yaml of application on existing GitRepo', { tags: '@fleet-41' }, () => {
+      const repoName = 'local-gitreporestrictions-fleet-41'
+
+      // Create GitRepoRestrictions with allowedTargetNamespace
+      cy.accesMenuSelection('Continuous Delivery', 'Advanced', 'GitRepoRestrictions');
+      cy.clickButton('Create from YAML');
+      cy.readFile('assets/git-repo-restrictions-allowed-target-ns.yaml').then((content) => {
+        cy.get('.CodeMirror').then((codeMirrorElement) => {
+          const cm = (codeMirrorElement[0] as any).CodeMirror;
+          cm.setValue(content);
+        });
+      })
+      cy.clickButton('Create');
+
+      // Add Fleet repository and create it
+      cy.fleetNamespaceToggle('fleet-local');
+      cy.addFleetGitRepo({repoName, repoUrl, branch, path});
+      cy.clickButton('Create');
+      cy.verifyTableRow(0, 'Error', repoName);
+      cy.get('table > tbody > tr.state-description[data-testid="sortable-table-0-row-description"] > td.text-error')
```

Comment:
Pls try to avoid absolute/semi-absolute path locators.
This works:
```
tr.state-description[data-testid="sortable-table-0-row-description"] > td.text-error
```
also this:
```
td.text-error
```

Reply:
Initially, I thought about this locator: `td.text-error` but later I used the whole one.
I will update with second one. Thanks :smile: